staging/src/k8s.io/apimachinery/pkg/util/proxy/transport_test.go

```diff
@@ -274,3 +277,65 @@ func TestProxyTransport(t *testing.T) {
 		testItem(name, &item)
 	}
 }
+
+func TestRewriteResponse(t *testing.T) {
+	gzipbuf := bytes.NewBuffer(nil)
+	flatebuf := bytes.NewBuffer(nil)
+	gzw := gzip.NewWriter(gzipbuf)
+	flw, _ := flate.NewWriter(flatebuf, flate.BestCompression)
+	defer gzw.Close()
+	defer flw.Close()
+
+	testTransport := &Transport{
+		Scheme:      "http",
+		Host:        "foo.com",
+		PathPrepend: "/proxy/node/node1:10250",
+	}
+
+	encodeType := []string{"gzip", "deflate"}
+	expectedGzip := "gzip test"
+	expectedDeflate := "deflate test"
+	gzw.Write([]byte(expectedGzip))
+	gzw.Flush()
+	flw.Write([]byte(expectedDeflate))
+	flw.Flush()
+	resp := []http.Response{
+		{
+			Body: ioutil.NopCloser(gzipbuf),
+		},
+		{
+			Body: ioutil.NopCloser(flatebuf),
+		},
+	}
+	errFn := func(encode string, err error) {
+		t.Errorf("%s failed to read and write: %v", encode, err)
+	}
+	for k, v := range encodeType {
+		request, err := http.NewRequest("GET", "http://mynode.com/", nil)
+		request.Header.Set("Content-Encoding", v)
+		request.Header.Add("Accept-Encoding", v)
+
+		gotResponse, err := testTransport.rewriteResponse(request, &resp[k])
+
+		if err != nil {
+			errFn(v, err)
+		}
+
+		switch v {
+		case "deflate":
+			s, _ := ioutil.ReadAll(flate.NewReader(gotResponse.Body))
```

Comment:
Package a reader factory function in the table, too. We shouldn't have switches like this inside the loop.